src/System.Security.Cryptography.Cng/src/Resources/Strings.resx

```diff
@@ -0,0 +1,186 @@
+  </data>
+  <data name="Cryptography_ArgECDsaRequiresECDsaKey" xml:space="preserve">
+    <value>Keys used with the ECDsaCng algorithm must have an algorithm group of ECDsa.</value>
+  </data>
+  <data name="Cryptography_ArgExpectedECDiffieHellmanCngPublicKey" xml:space="preserve">
+    <value>DeriveKeyMaterial requires an ECDiffieHellmanCngPublicKey.</value>
+  </data>
+  <data name="Cryptography_ArgRSAaRequiresRSAKey" xml:space="preserve">
+    <value>Keys used with the RSACng algorithm must have an algorithm group of RSA.</value>
+  </data>
+  <data name="Cryptography_InvalidAlgorithmGroup" xml:space="preserve">
+    <value>The algorithm group '{0}' is invalid.</value>
+  </data>
+  <data name="Cryptography_InvalidAlgorithmName" xml:space="preserve">
+    <value>The algorithm name '{0}' is invalid.</value>
+  </data>
+  <data name="Cryptography_InvalidKeyBlobFormat" xml:space="preserve">
+    <value>The key blob format '{0}' is invalid.</value>
+  </data>
+  <data name="Cryptography_InvalidProviderName" xml:space="preserve">
+    <value>The provider name '{0}' is invalid.</value>
+  </data>
+  <data name="Cryptography_InvalidRsaParameters" xml:space="preserve">
+    <value>The specified RSA parameters are not valid; both Exponent and Modulus are required fields.</value>
+  </data>
+  <data name="Cryptography_OpenEphemeralKeyHandleWithoutEphemeralFlag" xml:space="preserve">
+    <value>The CNG key handle being opened was detected to be ephemeral, but the EphemeralKey open option was not specified.</value>
+  </data>
+  <data name="Cryptography_OpenInvalidHandle" xml:space="preserve">
+    <value>Cannot open an invalid handle.</value>
+  </data>
+  <data name="NotSupported_Method" xml:space="preserve">
+    <value>Method not supported.</value>
+  </data>
+  <data name="Cryptography_NotValidPrivateKey" xml:space="preserve">
+    <value>Key is not a valid private key.</value>
+  </data>
+  <data name="Cryptography_NotValidPublicOrPrivateKey" xml:space="preserve">
+    <value>Key is not a valid public or private key.</value>
+  </data>
+  <data name="Cryptography_InvalidSignatureAlgorithm" xml:space="preserve">
+    <value>The hash algorithm is not supported for signatures. Only MD5, SHA1, SHA256,SHA384, and SHA512 are supported at this time.</value>
+  </data>
+  <data name="Cryptography_NonCompliantFIPSAlgorithm" xml:space="preserve">
+    <value>This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.</value>
+  </data>
+  <data name="Cryptography_TlsRequiresLabelAndSeed" xml:space="preserve">
+    <value>The TLS key derivation function requires both the label and seed properties to be set.</value>
+  </data>
+  <data name="Cryptography_UnsupportedPaddingMode" xml:space="preserve">
+    <value>The specified PaddingMode is not supported.</value>
+  </data>
+  <data name="Cryptography_ECXmlSerializationFormatRequired" xml:space="preserve">
```

Comment:
We can prune them after the library is complete. For productivity efficiency, I hoovered up every string I could find the CNG-related desktop code. Nothing more annoying than being yanked out of the zone because of a missing SR string.



In src/System.Security.Cryptography.Cng/src/Resources/Strings.resx<https://github.com/dotnet/corefx/pull/2615#discussion_r36243957>:

> +  <data name="Cryptography_NotValidPublicOrPrivateKey" xml:space="preserve">

> +    <value>Key is not a valid public or private key.</value>

> +  </data>

> +  <data name="Cryptography_InvalidSignatureAlgorithm" xml:space="preserve">

> +    <value>The hash algorithm is not supported for signatures. Only MD5, SHA1, SHA256,SHA384, and SHA512 are supported at this time.</value>

> +  </data>

> +  <data name="Cryptography_NonCompliantFIPSAlgorithm" xml:space="preserve">

> +    <value>This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.</value>

> +  </data>

> +  <data name="Cryptography_TlsRequiresLabelAndSeed" xml:space="preserve">

> +    <value>The TLS key derivation function requires both the label and seed properties to be set.</value>

> +  </data>

> +  <data name="Cryptography_UnsupportedPaddingMode" xml:space="preserve">

> +    <value>The specified PaddingMode is not supported.</value>

> +  </data>

> +  <data name="Cryptography_ECXmlSerializationFormatRequired" xml:space="preserve">

Do we even have EC XML Serialization? Are the strings here the minimal required set?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/2615/files#r36243957>.